### ISSUE TITLE
HDF5 version problem on windows

### ISSUE THREAD

**Issue by twiecki** [conda-forge/pymc3-feedstock#19]
@wtpfede writes:

Hi all,
I just upgraded to the last version of pyMC3 using:

> conda install -c conda-forge pymc3

Unfortunately, it seems like this particular version of pyMC3 has issues interfacing with the HDF5 library.
I have already upgraded to the latest version of HDF5, via:

> conda install -c anaconda hdf5=1.8.17

but this has not solved the problem. The crash dump (enclosed below) points to a version mismatch, but there's no 1.8.18 version of hdf5 available (yet?)

I'm on windows 8, python 3.5, anaconda 4.2.13, and I have (for now) rolled back to pymc3 3.0 (which is working fine). I'm also not having the same issue on Linux.

Anyone else experiencing the same problem and/or has a fix for this?

This is what the console outputs right before crashing:

```
Warning! ***HDF5 library version mismatched error***
The HDF5 header files used to compile this application do not match
the version used by the HDF5 library to which this application is linked.
Data corruption or segmentation faults may occur if the application continues.
This can happen when an application was compiled by one version of HDF5 but
linked with a different version of static or shared HDF5 library.
You should recompile the application or check your shared library related
settings such as 'LD_LIBRARY_PATH'.
You can, at your own risk, disable this warning by setting the environment
variable 'HDF5_DISABLE_VERSION_CHECK' to a value of '1'.
Setting it to 2 or higher will suppress the warning messages totally.
Headers are 1.8.17, library is 1.8.18

      SUMMARY OF THE HDF5 CONFIGURATION
      =================================
```
General Information:
-------------------
                   HDF5 Version: 1.8.18
                  Configured on: 2017-03-06
                  Configured by: Visual Studio 14 2015 Win64
                 Configure mode: CMAKE 3.7.2
                    Host system: Windows-10.0.14393
              Uname information: Windows
                       Byte sex: little-endian
                      Libraries:
             Installation point: C:/Program Files/HDF5

Compiling Options:
------------------

```
               Compilation Mode:
                     C Compiler: C:/Program Files (x86)/Microsoft Visual Studio
14.0/VC/bin/x86_amd64/cl.exe
                         CFLAGS: /DWIN32 /D_WINDOWS /W3
                      H5_CFLAGS:
                      AM_CFLAGS:
                       CPPFLAGS:
                    H5_CPPFLAGS:
                    AM_CPPFLAGS:
               Shared C Library: YES
               Static C Library: YES
  Statically Linked Executables: OFF
                        LDFLAGS: /machine:x64
                     AM_LDFLAGS:
                Extra libraries: X:inclib-vc14-x64/zlib.lib;X:/inclib-vc14-x64/l
ibsz.lib;X:/inclib-vc14-x64/libaec.lib
                       Archiver:
                         Ranlib:
              Debugged Packages:
                    API Tracing: OFF

```

Languages:
----------
                        Fortran: OFF
               Fortran Compiler:
          Fortran 2003 Compiler:
                  Fortran Flags:
               H5 Fortran Flags:
               AM Fortran Flags:
         Shared Fortran Library: YES
         Static Fortran Library: YES
```

                            C++: ON
                   C++ Compiler: C:/Program Files (x86)/Microsoft Visual Studio
14.0/VC/bin/x86_amd64/cl.exe
                      C++ Flags: /DWIN32 /D_WINDOWS /W3 /GR /EHsc
                   H5 C++ Flags:
                   AM C++ Flags:
             Shared C++ Library: YES
             Static C++ Library: YES
```

Features:
---------
```
                  Parallel HDF5: OFF
             High Level library: ON
                   Threadsafety: OFF
            Default API Mapping: v18
 With Deprecated Public Symbols: ON
         I/O filters (external):  DEFLATE DECODE ENCODE
                            MPE:
                     Direct VFD:
                        dmalloc:
Clear file buffers before write: ON
           Using memory checker: OFF
         Function Stack Tracing: OFF
      Strict File Format Checks: OFF
   Optimization Instrumentation:
Bye...
```

**Comment by twiecki** [comment]
@ericmjl @jakirkham any idea? Must be the `h5py` dependency which is probably optional.

**Comment by ericmjl** [comment]
@twiecki not quite sure here. 

@jakirkham some hypotheses:

- Remove`h5py` dependency?
- Add `hdf5` dependency and pin it?

Admittedly, this is the first time I've seen this issue, so I'm not quite sure I'm thinking in the right way about it.

**Comment by jakirkham** [comment]
Not sure what I'm being asked here or what the issue is. That said, the latest `hdf5` is 1.8.18 and `h5py` 2.7.0 has already been rebuilt with it. Both exist on Windows (just checked the `conda-forge` channel). Why is `hdf5` 1.8.17 being used? I expect the answer is buried in the user's environment, which doesn't appear to be provided. In any event, don't have time this week to investigate this further.

**Comment by wtpfede** [comment]
Hi all,
after some more poking around tonight, it seemed indeed like the problem was simply with my environment.

I just downgraded hdf5 to the 1.8.15 version via:

> conda install -c anaconda hdf5=1.8.15

And i uninstalled a double (weird) installation of h5py via 

> conda uninstall h5py
and
> pip uninstall h5py

finally, I upgraded to the latest pymc3 version again by

> conda install -c conda-forge pymc3=3.1

Works perfectly now. 
Thanks for your time and sorry for bothering you!
